lib/dogwatch/model/options.rb

```diff
@@ -55,6 +64,24 @@ def notify_audit(notify = false)
       def include_tags(include = true)
         @attributes.include_tags = !!include
       end
+
+      # @param [Hash{String=>Fixnum}] thresholds
+      def thresholds(thresholds)
+        validate_monitor_type_specific_option!(:thresholds)
+        @attributes.thresholds = thresholds
+      end
+
+      private
+
+      def validate_monitor_type_specific_option!(option)
+        options = Array(MONITOR_TYPE_OPTIONS_MAP[@monitor_type])
+        return true if options.include?(option)
+
+        # rubocop:disable Metrics/LineLength
+        message = "The #{@monitor_type.inspect} monitor type does not support #{option.inspect}."
+        message << " Did you mean one of #{options.join(', ')}?" if options.any?
+        raise NotImplementedError, message
```

Comment:
@bturner-r7 right now this error message will state "`The nil monitor type...`". Dogwatch does require the type to be set later on where validation fails if type unless is a non-nil/non-empty string. Since we always document that `type` should be called and I've made it so that calling it will set monitor type appropriately I figured the error is sufficient.

>README.md:
>```ruby
>type :metric_alert # REQUIRED: One of [:metric_alert | :service_check | :event_alert]
>```